lib/Dialect/SPIRV/Serialization/Serializer.cpp

```diff
@@ -558,6 +569,21 @@ Serializer::processGlobalVariableOp(spirv::GlobalVariableOp varOp) {
   if (failed(processType(varOp.getLoc(), varOp.type(), resultTypeID))) {
     return failure();
   }
+
+  if (isStorageBufferType(varOp.type())) {
+    auto structType = varOp.type()
+                          .cast<spirv::PointerType>()
+                          .getPointeeType()
+                          .cast<spirv::StructType>();
+    SmallVector<uint32_t, 2> args{
+        findTypeID(structType),
+        static_cast<uint32_t>(spirv::Decoration::Block)};
+    if (failed(encodeInstructionInto(decorations, spirv::Opcode::OpDecorate,
+                                     args))) {
+      return emitError(varOp.getLoc(), "cannot decorate ") << structType;
```

Comment:
Do not use the global emitError if you have an op instance, use the 'op.emitError' methods instead:
varOp.emitError

Reply:
Thanks, fixed.